spring-cloud-dataflow-server-core/src/main/java/org/springframework/cloud/dataflow/server/controller/ComposedTaskController.java

```diff
@@ -0,0 +1,167 @@
+	 *     <li>task status checks to the provided {@link TaskLauncher}</li>
+	 * </ul>
+	 *
+	 * @param repository the repository this controller will use for task CRUD operations.
+	 * @param deploymentIdRepository the repository this controller will use for deployment IDs
+	 * @param taskLauncher the TaskLauncher this controller will use to check task status.
+	 * @param composedTaskProperties properties used to get composed task information.
+	 */
+	public ComposedTaskController(TaskDefinitionRepository repository, DeploymentIdRepository deploymentIdRepository,
+			TaskLauncher taskLauncher,ComposedTaskProperties composedTaskProperties) {
+		Assert.notNull(repository, "repository must not be null");
+		Assert.notNull(deploymentIdRepository, "DeploymentIdRepository must not be null");
+		Assert.notNull(taskLauncher, "taskLauncher must not be null");
+		Assert.notNull(composedTaskProperties, "composedTaskProperties must not be null");
+		this.repository = repository;
+		this.deploymentIdRepository = deploymentIdRepository;
+		this.taskLauncher = taskLauncher;
+		this.composedTaskProperties = composedTaskProperties;
+	}
+
+	/**
+	 * Register a composed task definition for future execution.
+	 *
+	 * @param name name of the composed task
+	 * @param graph DSL definition for the composed task
+	 */
+	@RequestMapping(value = "/compose", method = RequestMethod.POST)
+	public ComposedTaskResource saveComposedTask(@RequestParam("name") String name,
+			@RequestParam("definition") String graph) {
+		TaskDefinition taskDefinition = new TaskDefinition(name, graph, true);
+		TaskDefinition composedTaskDefinition = new TaskDefinition(
+				taskDefinition.getName(),getComposedArgs(taskDefinition.getDslText()));
+		validateComposedTaskElementsExist(name, graph);
+		repository.save(composedTaskDefinition);
+		return taskAssembler.toResource(taskDefinition);
+	}
+
+	private void validateComposedTaskElementsExist(String name, String graph) {
+		ComposedTaskValidator composedTaskValidator =
+				new ComposedTaskValidator(repository);
+		ComposedTaskNode composedTaskNode = new ComposedTaskParser().parse(name, graph);
+		composedTaskNode.accept(composedTaskValidator);
+		if(composedTaskValidator.getInvalidDefinitions().size() > 0) {
+			List<ComposedTaskValidationProblem> composedTaskValidationProblems =
+					composedTaskValidator.getInvalidDefinitions()
+							.values().stream()
+							.map(pos -> new ComposedTaskValidationProblem(
+									graph, pos, DSLMessage.CT_ELEMENT_IN_COMPOSED_DEFINITION_DOES_NOT_EXIST))
+							.collect(Collectors.toList());
+			if (!composedTaskValidationProblems.isEmpty()) {
+				throw new ComposedTaskValidationException(
+						composedTaskNode, composedTaskValidationProblems);
+			}
+		}
+	}
+
+	private String getComposedArgs(String graph) {
+		return String.format("%s --graph=\"%s\"", composedTaskProperties.getTaskName(), graph.replaceAll("\"", "\\\\\""));
```

Comment:
Extracting a method named `escapeQuotes` or something would also clarify the intent